src/service/database/databaseservice.ts

```diff
@@ -14,16 +14,29 @@ import { QueryOptions } from "./queryoptions";
 import { UpdateOptions } from "./updateoptions";
 import { UpdateResult } from "./updateresult";
 import { DeleteOptions } from "./deleteoptions";
-
+import { DatabaseEncryption } from "./database_encryption";
+
+/**
+ * Database service is for save JSON data on the mongo database on hive node.
+ *
+ * It also support the encryption of the fields of documents.
+ *
+ * If wants encryptDoc the document, please make sure the document is Object or Map.
+ */
 export class DatabaseService extends RestService {
 	private static LOG = new Logger("DatabaseService");
 
 	private static API_COLLECTION_ENDPOINT = "/api/v2/vault/db/collection";
 	private static API_COLLECTIONS_ENDPOINT = "/api/v2/vault/db/collections";
 	private static API_DB_ENDPOINT = "/api/v2/vault/db";
 
-    constructor(serviceContext: ServiceEndpoint, httpClient: HttpClient) {
+	private readonly encrypt: boolean;
+	private readonly databaseEncrypt: DatabaseEncryption;
+
+    constructor(serviceContext: ServiceEndpoint, httpClient: HttpClient, encrypt: boolean = false) {
 		super(serviceContext, httpClient);
+        this.encrypt = encrypt;
```

Comment:
Maybe it's better to define a new DatabaseService, like EncryptedDatabaseService, which would be exteneded from DatabaseService class.